internal/config/settings.go

```diff
@@ -312,6 +312,19 @@ func InitLocalSettings(settings map[string]interface{}, path string) {
 	}
 }
 
+func WriteJson(filename string, v interface{}) error {
+		file, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE, 0644)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		file.Truncate(0)
```

Comment:
This is not related with the pull request and I just remembered an old issue, but is `file.Truncate(0)` done instead of passing `O_TRUNC` to `os.OpenFile` as a workaround with something like #2660?

Reply:
No, that was just my unfamiliarity with Go. So this here is better:

```suggestion
		file, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
		if err != nil {
			return err
		}
		defer file.Close()
```